### DISCUSSION TITLE
Evaluating function calling functions

### DISCUSSION THREAD

**Discussion by timrogers** [microsoft/promptflow#1815] [Q&A] [answered]
Hi there!

I’m building applications that use OpenAI’s support for function calling. 

In the same way as you might test multiple variants of a prompt to find the prompt that works best, I’m looking for a way to test multiple variants of a function, since functions also need to be “prompt engineered” and o work well. 

Is that something that Prompt Flow can support? I didn’t find anything relevant in the docs.

**Comment by D-W-** [comment, marked as answer]
Hi @timrogers , it's supported to tune function calling with variants. Just like other variants, you can create different variants for same LLM node with different function inputs like this: 
![image](https://github.com/microsoft/promptflow/assets/7776147/15001f05-02e3-4dbd-b397-7da447be0c4b)

**Reply by timrogers** [reply]
@D-W- Thanks for taking the time to get back to me! This is really helpful. Is there a pattern you would recommend for defining function implementations?

**Reply by D-W-** [reply]
Currently we don't have best practices for defining function implementations. But we'll soon add an example on how to use function calling with prompt flow. I'll update here once we added the example.

**Reply by timrogers** [reply]
Thanks! That sounds great ❤️ Do you have a rough timeline in mind for that?

**Reply by D-W-** [reply]
@YingChen1996 please help provide an ETA here

**Reply by YingChen1996** [reply]
have opened a task to track migrate the function calling sample to prompt-flow repo and the ETA is 2/2.

**Reply by YingChen1996** [reply]
Function call samples have been merged: https://github.com/microsoft/promptflow/tree/main/examples/flows/chat/use_functions_with_chat_models
